org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/ClassFileEditor.java

```diff
@@ -649,11 +651,19 @@ protected IEditorInput transformEditorInput(IEditorInput input) {
 		return input;
 	}
 
+	@Override
+	public void init(IEditorSite site, IEditorInput input) throws PartInitException {
+		JavaCore.run(() -> super.init(site, input));
+	}
 	/*
 	 * @see AbstractTextEditor#doSetInput(IEditorInput)
 	 */
 	@Override
 	protected void doSetInput(IEditorInput input) throws CoreException {
+//		JavaCore.run(() -> doSetInputCached(input));
```

Comment:
Why is this commented?

Reply:
i was trying around a bit. obviously the hierarchy views have the most benefit as they typically reference multiple classes, but also a lot of other UI actions call jdt.core multiple times. For example both ClassFileEditor.doSetInput(), selectionChanged(), createPartControl().. are called after each other. But there is not common JDT UI call around.
Best would be if one could inject on a general ui layer on the callstack like
```
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:728)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:712)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.lambda$0(PartServiceImpl.java:105)
	at org.eclipse.e4.ui.services.internal.events.UIEventHandler.lambda$0(UIEventHandler.java:38)
	at org.eclipse.swt.widgets.Synchronizer.syncExec(Synchronizer.java:183)
	at org.eclipse.ui.internal.UISynchronizer.syncExec(UISynchronizer.java:133)
	at org.eclipse.swt.widgets.Display.syncExec(Display.java:4802)
	at org.eclipse.e4.ui.workbench.swt.DisplayUISynchronize.syncExec(DisplayUISynchronize.java:34)
	at org.eclipse.e4.ui.services.internal.events.UIEventHandler.handleEvent(UIEventHandler.java:38)
	at org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent(EventHandlerWrapper.java:208)
```
but i don't know how. Any idea?